main.go

```diff
@@ -106,17 +107,25 @@ func RunManager(c *cli.Context) error {
 			EngineImage: engineImage,
 			Network:     c.String(FlagDockerNetwork),
 		}
-		docker, err := docker.NewDockerOrchestrator(cfg)
+		orch, err = docker.NewDockerOrchestrator(cfg)
+		if err != nil {
+			return err
+		}
+	} else if orchName == "kubernetes" {
+		cfg := &kubernetes.Config{
+			EngineImage: engineImage,
+		}
+		orch, err = kubernetes.NewKuberOrchestrator(cfg)
 		if err != nil {
 			return err
 		}
-
-		forwarder = orchestrator.NewForwarder(docker)
-		orch = forwarder
 	} else {
 		return fmt.Errorf("invalid orchestrator %v", orchName)
 	}
 
+	forwarder = orchestrator.NewForwarder(orch)
```

Comment:
`orch = orchestrator.NewForwarder(orch)`

Reply:
orch pointer will be lost.
    orchestrator.Orchestrator is different from  orchestrator.Forwarder

Reply:
orchestrator.Orchestrator has no method SetLocator and

Reply:
OK, I saw the reference below.